test/coe.lean

```diff
@@ -0,0 +1,45 @@
+import Std.Tactic.CoeExt
+import Std.Tactic.GuardMsgs
+
+set_option linter.missingDocs false
+
+structure WrappedNat where
+  val : Nat
+
+
+structure WrappedFun where
+  fn : Nat → Nat
+
+
+structure WrappedType where
+  typ : Type
+
+attribute [coe] WrappedNat.val
+instance : Coe WrappedNat Nat where coe := WrappedNat.val
+
+#eval Std.Tactic.Coe.registerCoercion ``WrappedFun.fn (some ⟨1, 0, .coeFun⟩)
+instance : CoeFun WrappedFun (fun _ => Nat → Nat) where coe := WrappedFun.fn
+
+#eval Std.Tactic.Coe.registerCoercion ``WrappedType.typ (some ⟨1, 0, .coeSort⟩)
+instance : CoeSort WrappedType Type where coe := WrappedType.typ
+
+
+variable (n : WrappedNat) (f : WrappedFun) (t : WrappedType)
+
+/-- info: ↑n : Nat -/
+#guard_msgs in #check n.val
+/-- info: ↑n : Nat -/
+#guard_msgs in #check (↑n : Nat)
+
+/-- info: ⇑f : Nat → Nat -/
+#guard_msgs in #check f.fn
+/-- info: ⇑f : Nat → Nat -/
+#guard_msgs in #check ⇑f
+-- applied functions do not need the `⇑`
+/-- info: f 1 : Nat -/
+#guard_msgs in #check ⇑f 1
```

Comment:
Can you add tests for some more complicated cases, like 
```
fun f : Nat -> WrappedFun => f 2
fun f : Nat -> WrappedFun => f 2 3
``` 
and a test of a term that would in Lean 3 be written as `⇑(⇑f x) y`

Reply:
Done; the output is `(f x) y`, which isn't perfect but is still an improvement over what we currently have.